llvm-build/libllvm.ps1

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+. (Join-Path $PSScriptRoot .. build "set-env.ps1")
+. (Join-Path $PSScriptRoot "utils.ps1")
+
+Assert (Test-Path env:\PKG_NAME) "PKG_NAME was not defined"
+Assert (Test-Path env:\LINUX_PKG_NAME) "LINUX_PKG_NAME was not defined"
+Assert (Test-Path env:\WINDOWS_PKG_NAME) "WINDOWS_PKG_NAME was not defined"
+Assert (Test-Path env:\DARWIN_PKG_NAME) "DARWIN_PKG_NAME was not defined"
+Assert (Test-Path env:\BLOBS_OUTDIR) "BLOBS_OUTDIR was not defined"
+Assert (Test-Path env:\BUILD_ARTIFACTSTAGINGDIRECTORY) "BUILD_ARTIFACTSTAGINGDIRECTORY was not defined"
+
+$llvmArchiveFiles = @(
+    "$($env:LINUX_PKG_NAME)/llvm-build/llvm-project/build/$($env:LINUX_PKG_NAME).tar.gz",
+    "$($env:WINDOWS_PKG_NAME)/llvm-build/llvm-project/build/$($env:WINDOWS_PKG_NAME).zip",
+    "$($env:DARWIN_PKG_NAME)/llvm-build/llvm-project/build/$($env:DARWIN_PKG_NAME).tar.gz"
+)
+
+for ($i = 0; $i -lt $llvmArchiveFiles.Count; $i++) {
+    $file = $llvmArchiveFiles[$i]
+    Write-Vso "Raw: $file"
+    $path = Join-Path (Get-Location) '..' $file
+    Write-Vso "Joined: $path"
+    $resolvedPath = Resolve-Path $path
+    Write-Vso "Resolved: $resolvedPath"
+    $llvmArchiveFiles[$i] = $resolvedPath
+}
+
+$artifactsDir = $env:BLOBS_OUTDIR
+if (!(Test-Path $artifactsDir)) {
+    mkdir $artifactsDir | Out-Null
+}
+$artifactsDir = Resolve-Path "$artifactsDir"
+
+foreach ($file in $llvmArchiveFiles) {
+    $file = (Get-Item $file)
+    $extension = $file.Extension
+    if (".zip" -eq $extension) {
+        exec { 7z x $file }
+    }
+    elseif (".gz" -eq $extension) {
+        $tarGz = $file
+        Write-Vso "Processing $tarGz"
+        exec { 7z x $tarGz }
+        $tar = $tarGz.Name.TrimEnd($extension)
+        Write-Vso "Processing $tar"
+        exec { 7z x -aoa -ttar $tar }
+    }
```

Comment:
I assume there is no single `7z` command to extract `.tar.gz` (two separate commands are _required_).

Reply:
Yes, unfortunately it's an archive in an archive, so the two calls are required.